src/Core/Initialization/IInitializationProcessor.cs

```diff
@@ -20,19 +20,21 @@
 
 namespace SonarLint.VisualStudio.Core.Initialization;
 
+public interface IInitializationProcessorFactory
+{
+    IInitializationProcessor Create<T>(IReadOnlyCollection<IRequireInitialization> dependencies, Func<IThreadHandling, Task> initialization);
```

Comment:
Enforcing the delegate to receive `IThreadHandling` parameter leads to some additional boiler plate code in the implementation of the `IRequireInitialization`: we need to have two initialize methods one with the parameter and one without. If we simply remove it will simplify the code.

```suggestion
    IInitializationProcessor Create<T>(IReadOnlyCollection<IRequireInitialization> dependencies, Func<Task> initialization);
```

Reply:
The first option would not really work because the implementation of the lambda should never be directly accessible to other classes, meaning we'll still have 2 methods `public InitializeAsync(); + private InitializeImplementationAsync()`

I have considered the 2nd option and I decided to not change it because it felt like a big change. We can discuss that and I could make a separate PR with that.

Reply:
They were not two different options, but different proposals that can be done at the same time.
We don't need the `public InitializeAsync();` if we also do the second proposal.

> I have considered the 2nd option and I decided to not change it because it felt like a big change. We can discuss that and I could make a separate PR with that.
It is not really big. Just dropping a method from the interface. I think it's better do it as part of this PR as then we will have more places to change and the effort will be greater

Reply:
the 1stproposal does not make sense when the 2nd proposal is done. the 2nd proposal will cause more test changes and I already have a new commit based on this current PR

Reply:
I created a [follow up task](https://sonarsource.atlassian.net/browse/SLVS-2093).